internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -1167,6 +1187,16 @@ func upgradeSettingsForDataSourceSchema() *pluginsdk.Schema {
 					Type:     pluginsdk.TypeString,
 					Computed: true,
 				},
+				"drain_timeout_in_minutes": {
+					Type:     pluginsdk.TypeInt,
+					Optional: true,
+					Default:  "30",
+				},
+				"node_soak_duration_in_minutes": {
+					Type:     pluginsdk.TypeInt,
+					Optional: true,
+					Default:  0,
+				},
```

Comment:
This is the schema for the data source so these should only be `Computed`

```suggestion
				"drain_timeout_in_minutes": {
					Type:     pluginsdk.TypeInt,
					Computed: true,
				},
				"node_soak_duration_in_minutes": {
					Type:     pluginsdk.TypeInt,
					Computed: true,
				},
```